src/prefect/client/orchestration.py

```diff
@@ -2594,22 +2595,43 @@ async def send_worker_heartbeat(
         work_pool_name: str,
         worker_name: str,
         heartbeat_interval_seconds: Optional[float] = None,
-    ):
+        get_worker_id: bool = False,
+    ) -> Optional[UUID]:
         """
         Sends a worker heartbeat for a given work pool.
 
         Args:
             work_pool_name: The name of the work pool to heartbeat against.
             worker_name: The name of the worker sending the heartbeat.
+            return_id: Whether to return the worker ID.
```

Comment:
I think it'd be worthwhile to add a note here to explain why this kwarg is ignored in some cases.
```suggestion
            return_id: Whether to return the worker ID. Note: will return `None` if the connected server does not support returning worker IDs, even if `return_id` is `True`.
```